packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Primitives/CSharpType.cs

```diff
@@ -139,7 +139,9 @@ internal CSharpType(
             : this(
                   implementation.Name,
                   providerNamespace,
-                  implementation is EnumProvider || implementation.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Struct),
+                  implementation is EnumProvider ||
+                  implementation.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Struct) ||
```

Comment:
The struct condition seems wrong since having a struct modifier doesn't guarantee that it is an enum.